core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -32,18 +33,24 @@ import org.apache.spark.util.Utils
  * has started will events be actually propagated to all attached listeners. This listener bus
  * is stopped when `stop()` is called, and it will drop further events after stopping.
  */
-private[spark] class LiveListenerBus extends SparkListenerBus {
+private[spark] class LiveListenerBus(val sparkContext: SparkContext) extends SparkListenerBus {
```

Comment:
I'm modifying LiveListenerBus now and noticed that we're passing in `sparkContext` even though we only use it to access `conf`. I think it would have been better to just pass in `conf` here. It would make the initialization order constraints a lot clearer, too: right now it's not immediately clear why `eventQueue` needs to be a `lazy val`.